baybe/serialization/core.py

```diff
@@ -111,19 +65,15 @@ def structure_base(val: dict, _: Type[_T]) -> _T:
 
 
 def _structure_dataframe_hook(string: str, _) -> pd.DataFrame:
-    """De-serialize a DataFrame."""
-    buffer = BytesIO()
-    buffer.write(base64.b64decode(string.encode("utf-8")))
-    return pd.read_parquet(buffer)
+    """Deserialize a DataFrame."""
+    pickled_df = base64.b64decode(string.encode("utf-8"))
+    return pickle.loads(pickled_df)
```

Comment:
What is the reason for using pickle now?

Reply:
Simply because that whole dataframe serialization topic is just a nightmare 🙈 We started using human-readable stuff like csv but always got type inference problems in roundtrip serialization (e.g. ensure that a float remains a float after de-/serializing and is not converted to int, even when it's numeric value is 0.0). That's why – for the lack of better alternatives – I switched to binary representation at some point. But now the hypothesis test also discovered some edge cases for the parquet format I used ... All in all, this problem is not yet fully solved and will probably undergo more changes in the future, until we find a version that is both robust and ideally human-readable. So for now, just consider it as a quick fix, but feel free to look for better alternatives 🙃 Will resolve, but reactivate if further discussion is needed.